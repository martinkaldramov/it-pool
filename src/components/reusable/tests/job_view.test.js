import React from 'react';
import { shallow } from 'enzyme';
import JobView from '../job_view';
import mockJobs from '../../../mock_data/mock_jobs';

describe('JobView', () => {
  const job = mockJobs[0];
  const wrap = shallow(<JobView jobData={job} />);

  it('should render successfully', () => {
    expect(wrap).toMatchSnapshot();
  });
});
