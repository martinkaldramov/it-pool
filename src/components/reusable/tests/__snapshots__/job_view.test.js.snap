// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobView should render successfully 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MemoryRouter
    initialEntries={
        Array [
            "/jobs/1",
          ]
    }
>
    <JobView
        jobData={
            Object {
                "category": "Web Development",
                "city": "Sofia",
                "company": "MODIS BULGARIA",
                "date": 5,
                "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent sit amet facilisis ligula, ac consectetur nisl. Orci varius natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Fusce venenatis condimentum tristique. Quisque volutpat vulputate dolor, id rhoncus erat euismod id. Sed maximus dignissim arcu ac pretium. Curabitur vel ante at nunc sodales consectetur. Proin et porta sem, ac volutpat tellus. Curabitur ipsum lacus, gravida id fermentum id, pellentesque vitae erat.",
                "experience": "Solid",
                "id": 1,
                "skills": Array [
                  "React.JS",
                  "Angular.io",
                  "DevOps",
                  "ExpressJS",
                  "Google Firebase",
                ],
                "title": "Proofreading/light editing of a legal document translated from Japanese to English",
                "type": "Full Time",
                "views": 236,
              }
        }
    />
</MemoryRouter>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <JobView
        jobData={
                Object {
                        "category": "Web Development",
                        "city": "Sofia",
                        "company": "MODIS BULGARIA",
                        "date": 5,
                        "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent sit amet facilisis ligula, ac consectetur nisl. Orci varius natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Fusce venenatis condimentum tristique. Quisque volutpat vulputate dolor, id rhoncus erat euismod id. Sed maximus dignissim arcu ac pretium. Curabitur vel ante at nunc sodales consectetur. Proin et porta sem, ac volutpat tellus. Curabitur ipsum lacus, gravida id fermentum id, pellentesque vitae erat.",
                        "experience": "Solid",
                        "id": 1,
                        "skills": Array [
                          "React.JS",
                          "Angular.io",
                          "DevOps",
                          "ExpressJS",
                          "Google Firebase",
                        ],
                        "title": "Proofreading/light editing of a legal document translated from Japanese to English",
                        "type": "Full Time",
                        "views": 236,
                      }
        }
/>,
      "history": Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "30kn4g",
            "pathname": "/jobs/1",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "30kn4g",
          "pathname": "/jobs/1",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "jobData": Object {
          "category": "Web Development",
          "city": "Sofia",
          "company": "MODIS BULGARIA",
          "date": 5,
          "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent sit amet facilisis ligula, ac consectetur nisl. Orci varius natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Fusce venenatis condimentum tristique. Quisque volutpat vulputate dolor, id rhoncus erat euismod id. Sed maximus dignissim arcu ac pretium. Curabitur vel ante at nunc sodales consectetur. Proin et porta sem, ac volutpat tellus. Curabitur ipsum lacus, gravida id fermentum id, pellentesque vitae erat.",
          "experience": "Solid",
          "id": 1,
          "skills": Array [
            "React.JS",
            "Angular.io",
            "DevOps",
            "ExpressJS",
            "Google Firebase",
          ],
          "title": "Proofreading/light editing of a legal document translated from Japanese to English",
          "type": "Full Time",
          "views": 236,
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <JobView
          jobData={
                    Object {
                              "category": "Web Development",
                              "city": "Sofia",
                              "company": "MODIS BULGARIA",
                              "date": 5,
                              "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent sit amet facilisis ligula, ac consectetur nisl. Orci varius natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Fusce venenatis condimentum tristique. Quisque volutpat vulputate dolor, id rhoncus erat euismod id. Sed maximus dignissim arcu ac pretium. Curabitur vel ante at nunc sodales consectetur. Proin et porta sem, ac volutpat tellus. Curabitur ipsum lacus, gravida id fermentum id, pellentesque vitae erat.",
                              "experience": "Solid",
                              "id": 1,
                              "skills": Array [
                                "React.JS",
                                "Angular.io",
                                "DevOps",
                                "ExpressJS",
                                "Google Firebase",
                              ],
                              "title": "Proofreading/light editing of a legal document translated from Japanese to English",
                              "type": "Full Time",
                              "views": 236,
                            }
          }
/>,
        "history": Object {
          "action": "POP",
          "block": [Function],
          "canGo": [Function],
          "createHref": [Function],
          "entries": Array [
            Object {
              "hash": "",
              "key": "30kn4g",
              "pathname": "/jobs/1",
              "search": "",
              "state": undefined,
            },
          ],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "index": 0,
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "key": "30kn4g",
            "pathname": "/jobs/1",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "jobData": Object {
            "category": "Web Development",
            "city": "Sofia",
            "company": "MODIS BULGARIA",
            "date": 5,
            "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent sit amet facilisis ligula, ac consectetur nisl. Orci varius natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Fusce venenatis condimentum tristique. Quisque volutpat vulputate dolor, id rhoncus erat euismod id. Sed maximus dignissim arcu ac pretium. Curabitur vel ante at nunc sodales consectetur. Proin et porta sem, ac volutpat tellus. Curabitur ipsum lacus, gravida id fermentum id, pellentesque vitae erat.",
            "experience": "Solid",
            "id": 1,
            "skills": Array [
              "React.JS",
              "Angular.io",
              "DevOps",
              "ExpressJS",
              "Google Firebase",
            ],
            "title": "Proofreading/light editing of a legal document translated from Japanese to English",
            "type": "Full Time",
            "views": 236,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
